### PR TITLE
DEV-4229 predict options optional and added test for images

### PR DESCRIPTION
**Pull request by nkassis**

### CODE REVIEW

File: indico/queries/model_groups.py [hunk 1]
```diff
@@ -270,19 +272,39 @@ def process_response(self, response):
...
+    def _args_strings(self, **kwargs):
+        args = [k for k in self.query_args.keys() if kwargs.get(cc_to_snake(k))]
+
+        query_args_string = ",".join(f"${k}: {self.query_args[k]}" for k in args)
+        model_predict_args = ",".join(f"{k}: ${k}" for k in args)
+
+        query = self.query.replace("<QUERY_ARGS>", query_args_string)
+        query = query.replace("<MODEL_PREDICT_ARGS>", model_predict_args)
+
+        print(query)
```
Comment (jkgenser) [inline indico/queries/model_groups.py:297]: remove print.

File: tests/integration/queries/test_document.py [hunk 2]
```diff
@@ -116,3 +116,27 @@ def test_document_extraction_batched(indico):
...
+
+    extract = client.call(RetrieveStorageObject(job.result))
+
+    assert type(extract) == dict
+    assert "pages" in extract
+    image = extract["pages"][0]["image"]
+
+    image = client.call(RetrieveStorageObject(image))
+
+    assert image
```
Comment (madisonmay) [inline tests/integration/queries/test_document.py:142]: Feel like this final assert should be an instance check or something

File: indico/queries/model_groups.py (hunk 1 above)
Comment (madisonmay) [inline indico/queries/model_groups.py]: rogue print

File: indico/queries/model_groups.py [hunk 3]
```diff
@@ -270,19 +272,39 @@ def process_response(self, response):
...
         """
 
+    query_args = {
+        "modelId": "Int!",
+        "data": "[String]",
+        "predictOptions": "JSONString"
+    }
+
+    def _args_strings(self, **kwargs):
+        args = [k for k in self.query_args.keys() if kwargs.get(cc_to_snake(k))]
```
Comment (madisonmay) [inline indico/queries/model_groups.py:289]: Generally think we should prefer loud errors to silent failures but this is internal so less critical

**Pull request merged**